package.json

```
@@ -11,9 +11,14 @@
   "main": "./index.js",
   "types": "./index.d.ts",
   "exports": {
-    "require": "./index.js",
-    "import": "./index.mjs",
-    "types": "./index.d.ts"
+    "require": {
+      "types": "./index.d.ts",
+      "default": "./index.js"
+    },
+    "import": {
+      "types": "./index.d.mts",
+      "default": "./index.mjs"
+    }
```

alternatively this is possible:
```suggestion
    "require": "./index.js",
    "import": "./index.mjs"
```

You can still rely on "sibling lookup" for your declaration files - personally I prefer to include an explicit `types` condition though and since you already had it (although at the end and that's incorrect) I decided to include it for both import/require conditions here.

---

Based on https://twitter.com/atcb/status/1634653545407610880, I recommend to omit the `"types"` export.

I also recommend to have a `"default"` export condition that matches the package type. So in this case the `"require"` condition would be name `"default"`, and placed last.